### DISCUSSION TITLE
FakeIp混合模式下进入国产游戏问题

### DISCUSSION THREAD

**Discussion by EdWard0x** [vernesong/OpenClash#4246] [Q&A]
混合模式下UDP是tun模式，这模式下进入游戏对局非常慢，大概得需要个2分钟，进了一次之后后面就正常了，有大佬顺手指点下这是哪方面没调整好吗？

**Comment by fejeIaqbhbv** [comment]
换 redir-host 或加 [filter](https://wiki.metacubex.one/config/dns/#fake-ip-filter) 试试?

tun 一直有各种奇奇怪怪的bug，所以不一定是没调好

**Reply by EdWard0x** [reply]
redir-host是可以解决的。filter没去测试过，自我感觉没什么用。。。
因为fakerip是个特性，所以不是很想抛弃。
按照大佬你的说法，采用tun模式bug多的话，那TCP和UDP都推荐采用TProxy模式？

**Reply by fejeIaqbhbv** [reply]
> redir-host是可以解决的
> 因为fakerip是个特性，所以不是很想抛弃。

redir-host 能解决，那加 [fake-ip-filter](https://wiki.metacubex.one/config/dns/#fake-ip-filter) 就是有效的，但你得自己抓域名，少抓也无效
> 采用tun模式bug多的话，那TCP和UDP都推荐采用TProxy模式？

但是 tun 的转发效果好，tproxy 会降级。你如果无感可以用 tproxy

但你的问题主要是 fake-ip ，所以换不换 tproxy 无所谓

p.s. openclash 没有 tproxy 代理 ipv4 tcp 的功能

**Reply by EdWard0x** [reply]
哦是这样的，这个添加域名的方法我之前试过，在webUI里面能看到的域名我都加过，测试无效（也可能是链接关闭的太快，没看见导致漏域名了），目前我是劫持了v4 v6，但是仅代理了v4流量，所以v6请求在UI中是看不见的

**Reply by fejeIaqbhbv** [reply]
log 开 debug 看，log最全

数量多也可以 openclash-运行日志-内核日志-debug-清除日志-***开启游戏***-下载日志

排除非游戏设备的ip和你确定与游戏无关的域名，其余全加进去看效果

**Reply by EdWard0x** [reply]
好的，感谢大佬，我这周末尝试一下

**Reply by EdWard0x** [reply]
今天试了一些。还是不行。挑选域名的原理是看哪些域名走的漏网之鱼，我就把这些域名加入到过滤（都是从字面上认不出来所属的域名）。其他的后缀为qq.com的，发现已经自动匹配国内了。游戏为国服的瓦洛兰特、英雄联盟

**Reply by fejeIaqbhbv** [reply]
不只是漏网之鱼！

所有的，能显示在日志或连接信息里的，不在 fake-ip-filter 里的，都是

一般filter会直接`geosite:cn`

而且瓦我不清楚，但联盟我这没发现过，你确定第一次慢不是游戏设备问题？
